### PR TITLE
feat: add block by number and remove epoch accumulators

### PR DESCRIPTION
**Pull request by KolbyML**
### What was wrong?
implements  https://github.com/ethereum/portal-network-specs/pull/334
### How was it fixed?

by removing our uses of epoch accumulators as a content type

I reworked our peertests which used  epochaccumulators  using some test data Nick gave me to use other history types instead

### CODE REVIEW

**Comment by KolbyML** [comment]
@njgheorghita I would like your feedback on what we will do with the 5 tests which were added which uses `EpochAccumulators` in https://github.com/ethereum/trin/pull/1345 and https://github.com/ethereum/trin/pull/1341

With `EpochAccumulators` being removed which are the biggest content type, and the minimum storage requirement being 1mb. What do you think the best way forward would be

File: ethportal-peertest/src/scenarios/gossip.rs
Comment (njgheorghita) [inline ethportal-peertest/src/scenarios/gossip.rs:171]: comment out of date

File: ethportal-peertest/src/scenarios/gossip.rs
Comment (njgheorghita) [inline ethportal-peertest/src/scenarios/gossip.rs:350]: I don't think you actually need to store the header locally in every target. As long as it's available in any single node, it should be available to the others for validation.

File: ethportal-peertest/src/scenarios/gossip.rs
Comment (njgheorghita) [inline ethportal-peertest/src/scenarios/gossip.rs:94]: Sorry, but I realize now that the large content values I gave were not ideal, and it shows in the complexity of this test, making it difficult to follow what's going on...
I updated #1438 with new assets that should simplify this test... 
You should be able to write a test that's accurately testing the gossip-dropped feature using only the..
- receipts from block `15040641` (320kb)
- body from block  `15040708` (763kb)
- (and the 2 headers for validation)
As it stands, there's just too much going on and it's tough to follow the exact code paths that are being tested

File: ethportal-peertest/src/scenarios/gossip.rs
Comment (KolbyML) [reply ethportal-peertest/src/scenarios/gossip.rs:350]: I am confused, am I not just storing the header in target?

**Comment by KolbyML** [comment]
@njgheorghita ready for another look

File: ethportal-peertest/src/utils.rs
Comment (njgheorghita) [inline ethportal-peertest/src/utils.rs:232]: shouldn't clippy be complaining about this unwrap?

File: portal-bridge/src/api/execution.rs
Comment (njgheorghita) [inline portal-bridge/src/api/execution.rs:80]:
```rust
    HistoryContentKey, // BlockHeaderByHash
    HistoryContentKey, // BlockHeaderByNumber
```

File: portal-bridge/src/bridge/history.rs
Comment (njgheorghita) [inline portal-bridge/src/bridge/history.rs:355]: I'd also get rid of this function and just inline the relevant code since there's no more gossiping going on

File: portal-bridge/src/bridge/era1.rs
Comment (njgheorghita) [inline portal-bridge/src/bridge/era1.rs:253]: It doesn't seem like we're using the `epoch_hash` that's being returned as the first value anymore, so you could just update this util fn to return the `epoch_acc`

File: trin-history/src/validation.rs
Comment (njgheorghita) [inline trin-history/src/validation.rs:54]: Maybe it'd be useful to differentiate this error message from the one on line 35

File: trin-validation/src/header_validator.rs
Comment (njgheorghita) [inline trin-validation/src/header_validator.rs:261]: This test still seems somewhat useful to leave in. We still want to test that trin constructs & validates (and invalidates) these header with proofs...
I would just update this test (& `invalidate_invalid_proofs`) to lookup the epoch accs directly from disk and use `PreMergeAccumulator::construct_proof()` instead of the removed `PreMergeAccumulator::generate_proof()`

File: trin-history/src/validation.rs
Comment (njgheorghita) [inline trin-history/src/validation.rs:151]: Seems worthwhile at least adding a simple validate/invalidate test for `BlockHeaderWithNumber` here

File: ethportal-peertest/src/utils.rs
Comment (KolbyML) [reply ethportal-peertest/src/utils.rs:232]: This is in `ethportal-peertest`, I thought testing stuff didn't have that check

**Comment by KolbyML** [comment]
@njgheorghita ready for another look

**Review by njgheorghita — COMMENTED** [review]
Looks good! One last thing to consider before merging... we still have some number of epoch accumulators out there in the dbs of our network nodes. I think we want to drop all of those out before deploying this change. So I think we should hold off on merging this until we've done the filtering. I can dust off #1286 tomorrow and then once that's deployed / the filtering is done, we can merge this pr

**Review by njgheorghita — APPROVED** [review]
Moving ahead with the plan from https://github.com/ethereum/trin/pull/1286#issuecomment-2356427362, so this is good to go on my end after @ogenev comments are addressed

**Pull request merged**
